components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -188,6 +190,13 @@ open class GleanInternalAPI {
      * Initialize the core metrics internally managed by Glean (e.g. client id).
      */
     private fun initializeCoreMetrics(applicationContext: Context) {
+        // Since all of the ping_info properties are required, we can't
+        // use the normal metrics API to set them, since those work
+        // asynchronously, and there is a race condition between when they
+        // are set and the possible sending of the first ping upon startup.
+        // Therefore, this uses the lower-level internal storage engine API
+        // to set these metrics, which is synchronous.
```

Comment:
I discovered this because I updated the schema.  Currently on master, the `appBuild` and `appDisplayVersion` aren't mentioned in the schema we are checking against, so frequently in tests they are missing but it's not complaining.  (That's my bad for not remembering to update the schema git commit when these keys were added to the schema).  If you'd prefer this was a separate PR, let me know.